motionplan/constraint.go

```diff
@@ -126,58 +126,77 @@ func (c *constraintHandler) CheckConstraints(cInput *ConstraintInput) (bool, flo
 	return true, score, ""
 }
 
-// NewCollisionConstraint is a helper function for creating a collision Constraint that takes a frame and geometries
-// representing obstacles and interaction spaces and will construct a collision avoidance constraint from them.
 func NewCollisionConstraint(
 	frame referenceframe.Frame,
-	goodInput []referenceframe.Input,
-	obstacles map[string]spatial.Geometry,
+	fs referenceframe.FrameSystem,
+	worldState *referenceframe.WorldState,
+	observationInput map[string][]referenceframe.Input,
 	collisionSpecifications []*Collision,
 	reportDistances bool,
-) Constraint {
-	zeroVols, err := frame.Geometries(goodInput)
+) (Constraint, error) {
+	// extract inputs corresponding to the frame
+	var goodInputs []referenceframe.Input
+	var err error
+	switch f := frame.(type) {
+	case *solverFrame:
+		goodInputs, err = f.mapToSlice(observationInput)
```

Comment:
Ignoring for the moment that it's currently useless for external code to call this public function.

Rather than having this sort of type switch here, consider creating a private function that takes only a `solverFrame`, and update the caller to use that.

This public function can continue to exist and can retain the same type signature, and its function can be to wrap the given frame in a solverFrame, and then call the private function.

Reply:
On the other hand, this all should be private anyway no? My understanding was that you were going to rewrite large portions of the constraint system and that our choice to expose these methods was going to change later